pyOCD/flash/flash.py

```diff
@@ -211,16 +211,40 @@ def loadPageBuffer(self, bufferNumber, flashPtr, bytes):
         # transfer the buffer to device RAM
         self.target.writeBlockMemoryUnaligned8(self.page_buffers[bufferNumber], bytes)
 
+    def programPhrase(self, flashPtr, bytes):
+        """
+        Flash one page
+        """
+
+        # prevent security settings from locking the device
+        bytes = self.overrideSecurityBits(flashPtr, bytes)
+
+        # first transfer in RAM
+        self.target.writeBlockMemoryUnaligned8(self.begin_data, bytes)
+
+        # update core register to execute the program_page subroutine
+        result = self.callFunction(self.flash_algo['pc_program_page'], flashPtr, len(bytes), self.begin_data)
+
+        # check the return code
+        if result != 0:
+            logging.error('programPage(0x%x) error: %i', flashPtr, result)
+
     def getPageInfo(self, addr):
         """
         Get info about the page that contains this address
 
         Override this function if variable page sizes are supported
         """
+        region = self.target.getMemoryMap().getRegionForAddress(addr)
+
         info = PageInfo()
         info.erase_weight = DEFAULT_PAGE_ERASE_WEIGHT
         info.program_weight = DEFAULT_PAGE_PROGRAM_WEIGHT
-        info.size = self.flash_algo['page_size']
+        if region.isFlash and region.blocksize > 0:
+            info.size = region.blocksize
+        else:
+            inf.size = self.page_size
```

Comment:
You're right, the blocksize should never be 0. That check was mostly added for error checking. Really, we could probably just remove the 'page_size' key from the flash algo dict, and related attribute in Flash.

Just noticed a typo on line 246 because of your comment! Thanks :)